src/lib.rs

```diff
@@ -218,6 +238,40 @@ impl HidApi {
         }
     }
 
+    /// Open a HID device using libusb_wrap_sys_device.
+    pub fn wrap_sys_device(&self, sys_dev: isize, interface_num: i32) -> HidResult<HidDevice> {
+        if cfg!(not(any(
+            feature = "linux-static-libusb",
+            feature = "linux-shared-libusb"
+        ))) {
+            return Err(HidError::HidApiError {
+                message: String::from("wrap_sys_device is not implement"),
+            });
+        }
+
+        let device = unsafe {
+            ffi::hid_libusb_wrap_sys_device(sys_dev, interface_num)
+        };
+
+        if device.is_null() {
+            let e = Err(HidError::HidApiError {
```

Comment:
call the check_error() method instead, it does all that already for you.

match on its return, and then you can return both the Ok() and Err() variants as Err()